odoo/osv/expression.py

```diff
@@ -968,11 +970,19 @@ def push_result(query, params):
                         right = tuple(right)
 
                     unaccent = self._unaccent(field) if sql_operator.endswith('like') else lambda x: x
-
-                    left = unaccent(model._generate_translated_field(alias, left, self.query))
                     instr = unaccent('%s')
+                    # TODO VSC: returns something like 'instr' = dog
+                    # 1: '["dog","chien","..."]' --> This doesn't look like the most optimal index
+                    if need_wildcard:
+                        # TODO CWG: currently, inactive languages are not removed in DB and can be searched by using this function
+                        left = unaccent('jsonb_path_query_array("%s"."%s", \'$.*\')::text' % (alias, left))
+                    else:
+                        lang = model.env.lang or 'en_US'
```

Comment:
you must use a proper query parameter

```py
>>> env.user.with_context(lang="hello world").env.lang
'hello world'
```

PS: this is by no means a complete review, just a quick passing comment!

Reply:
In the last second commit [[SECURITY] security for lang], I validate `lang`
`assert env.user.with_context(lang="hello world").env.lang is None`

BTW if 'fr_FR' is inactive
`assert env.user.with_context(lang="fr_FR").env.lang is None`

But yes, it would be better if we can pass lang as a parameter

Reply:
> I validate lang

That does not look sufficient, because that check in `env.lang` is functionally un-necessary. This means it may get optimized out at any point, for any reason.

Please add at least:
- an explicit comment above the check explaining that's it's strictly required because the query generation code depends on it for security reasons (and perhaps a note about it being cached)
- a test that covers this validation

I can't see either in the current PR, and I think it's important enough to fix before merging. Sorry if I missed it.

@robodoo r-

Reply:
I doubt calling robodoo in review comments works, at least the dashboard page is not reflecting it